src/index.js

```diff
@@ -23,25 +23,29 @@ let fightState = false;
 let enemy = {
   name: 'The dragon',
   hp: 100,
-  atk: 10,
   def: 5,
+  attacks: new Map([
+    [{ message: 'The dragon breathes devastating fire!', atk: 15 }, 1],
+    [{ message: 'The dragon takes a swipe with its razor sharp claws.', atk: 10 }, 5],
+    [{ message: 'The dragon bites with its terrifying jaws.', atk: 12 }, 2],
+  ]),
```

Comment:
The [`Map`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) serves a similar purpose to objects, but with a few minor differences we don't have to get into too much. The main things, which are why I used it here over an object, are: 
1. Keys can be ANYTHING (on regular objects, keys can only be string or Symbol). Here, the keys are attack objects (values are how frequently this attack should be used).
2. Maps can be used like `for (const [key, value] of map)`, while objects cannot.

Reply:
I learned some thing new, thanks! I'll practice using Maps where it might be helpful then!

Questions:
1. Can we store this kind of information easily into a database?
2. Suppose in the future, we have an enemy that can use 10 different attacks. Does the fact that our current example dragon being able to use less than 10 different attacks cause problems in the future?

Reply:
Scratch that, yes it's very easy to put in a databse, and I see your `chooseWeighted()` below.